src/jit/importer.cpp

```diff
@@ -6958,7 +6958,7 @@ bool  Compiler::impMethodInfo_hasRetBuffArg(CORINFO_METHOD_INFO * methInfo)
     //
   #ifdef FEATURE_HFA
     // On ARM HFAs are returned in registers.
-    if (!info.compIsVarArgs && IsHfa(methInfo->args.retTypeClass))
+    if (!info.compIsVarArgs && !opts.compUseSoftFP && IsHfa(methInfo->args.retTypeClass))
     {
```

Comment:
So, you mean that in the same function we can have calls to varargs and non-varargs. Yes, I didn't think about that.

But do you mean that Compiler::IsHfa matters only for calling convention and we won't get anything else wrong if we switch it off for ARM softfp calling convention?